cocos/gfx/webgpu/webgpu-device.ts

```diff
@@ -0,0 +1,787 @@
+/*
+ Copyright (c) 2024 Xiamen Yaji Software Co., Ltd.
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+
+import { fetchUrl } from 'pal/wasm';
+import glslangUrl from 'external:emscripten/webgpu/glslang.wasm';
+import twgslUrl from 'external:emscripten/webgpu/twgsl.wasm';
+import glslangLoader from 'external:emscripten/webgpu/glslang.js';
+import twgslLoader from 'external:emscripten/webgpu/twgsl.js';
```

Comment:
What is the `external:` usage? I have never heard of.

Reply:
Is`glslang.wasm` `twgsl.wasm` a must? So WebGPU must depend on wasm?

Reply:
Engine-defined rules corresponding to the external folder in the engine directory

Reply:
glslang.wasm and twgsl.wasm are primarily used to convert GLSL to WGSL. Currently, no other methods have been discovered for achieving this conversion.